File: api/src/RadioactivityApp/Measurements/Measurement.cs
using InfluxDB.Client.Core;
using System.Text.Json.Serialization;
using MeasurementAttribute = InfluxDB.Client.Core.Measurement;

namespace RadioactivityApp.Measurements;

[Measurement("odl")]
public sealed class Measurement
{
    [Column("station_id", IsTag = true)]
    [JsonIgnore]
    public string? Id { get; set; }

    [Column("station_kenn", IsTag = true)]
    [JsonIgnore]
    public string? Kenn { get; set; }

    [Column("timestamp", IsTimestamp = true)]
    [JsonInclude]
    public DateTime Timestamp { get; set; }

    [Column("value")]
    [JsonInclude]
    public double Value { get; set; }

    [Column("cosmic_value")]
    [JsonInclude]
    public double? ValueCosmic { get; set; }

    [Column("terrestrial_value")]
    [JsonInclude]
    public double? ValueTerrestrial { get; set; }

    [Column("unit", IsTag = true)]
    [JsonInclude]
    public string? Unit { get; set; }

    [Column("nuclide", IsTag = true)]
    [JsonInclude]
    public string? Nuclide { get; set; }

    [Column("validated", IsTag = true)]
    [JsonInclude]
    public bool Validated { get; set; }
}

File: api-client/lib/src/models/measurement.dart
import 'package:meta/meta.dart';

@sealed
@immutable
@internal
class Measurement {
  final DateTime timestamp;
  final double value;
  final double? valueCosmic;
  final double? valueTerrestrial;
  final String? unit;
  final String? nuclide;
  final bool validated;

  Measurement({
    required this.timestamp,
    required this.value,
    this.valueCosmic,
    this.valueTerrestrial,
    this.unit,
    this.nuclide,
    required this.validated,
  });

  Measurement.fromJson(Map<String, dynamic> json)
      : timestamp = DateTime.parse(json['Timestamp'] as String),
        value = json['Value'] as double,
        valueCosmic = json['ValueCosmic'] as double?,
        valueTerrestrial = json['ValueTerrestrial'] as double?,
        unit = json['Unit'] as String?,
        nuclide = json['Nuclide'] as String?,
        validated = json['Validated'] as bool;
}
